s4e-backend/src/main/java/pl/cyfronet/s4e/ex/ErrorHandlerHelper.java

```diff
@@ -52,6 +52,30 @@
         return map;
     }
 
+    public Map<String, Object> toResponseMap(QueryException e) {
+        BindingResult bindingResult = e.getBindingResult();
+        val map = new LinkedHashMap<String, Object>();
+
+        // populate general errors
+        if (bindingResult.hasGlobalErrors()) {
+            map.put("__general__", bindingResult.getGlobalErrors().stream()
+                    .map(objectError -> getMessage(objectError))
+                    .collect(Collectors.toUnmodifiableList()));
+        }
+
+        // populate field errors
+        for (FieldError fieldError : bindingResult.getFieldErrors()) {
+            map.putIfAbsent(fieldError.getField(), new ArrayList<>());
+            ((List) map.get(fieldError.getField())).add(
+                    messageSource.getMessage(fieldError.getCode(),
+                    fieldError.getArguments(),
+                    LocaleContextHolder.getLocale()));
```

Comment:
What's wrong with the getMessage method? If you need to pass arguments, you can modify the call in getMessage. `MessageSource::getMessage` is overloaded, and one method has a `@Nullable` args argument.